tests/lib/test_openstack_floating_ip.py

```diff
@@ -0,0 +1,278 @@
+import datetime
+import unittest
+from unittest import mock
+from unittest.mock import MagicMock, patch
+
+from openstack_api.openstack_floating_ip import OpenstackFloatingIP
+
+
+class OpenstackFloatingIPTests(unittest.TestCase):
+    """
+    Runs various tests to ensure we are using the Openstack
+    Floating IP module in the expected way
+    """
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.mocked_connection = MagicMock()
+        with patch(
+            "openstack_api.openstack_floating_ip.OpenstackIdentity"
+        ) as identity_mock:
+            self.instance = OpenstackFloatingIP(self.mocked_connection)
+            self.identity_module = identity_mock.return_value
+
+        self.api = self.mocked_connection.return_value.__enter__.return_value
+
+        self.mock_fip_list = [
+            {
+                "id": "fipid1",
+                "name": "130.246.211.216",
+                "created_at": "2020-06-28T14:00:00Z",
+                "updated_at": "2020-06-28T14:00:00Z",
+                "status": "DOWN",
+            },
+            {
+                "id": "fipid2",
+                "name": "130.246.211.217",
+                "created_at": "2021-04-28T14:00:00Z",
+                "updated_at": "2021-06-28T14:00:00Z",
+                "status": "ACTIVE",
+            },
+            {
+                "id": "fipid3",
+                "name": "130.246.212.218",
+                "created_at": "2021-06-28T14:00:00Z",
+                "updated_at": "2021-06-28T14:00:00Z",
+                "status": "ACTIVE",
+            },
+        ]
+
+    def test_search_all_fips_no_project(self):
+        """
+        Tests calling search_all_fips with no project selected
+        """
+        self.instance.search_all_fips(cloud_account="test", project_identifier="")
+
+        self.mocked_connection.assert_called_once_with("test")
+
+        self.api.list_floating_ips.assert_has_calls(
```

Comment:
Could we use `assert_called_once_with` instead?

Reply:
Yep, done